enhancements/20190903-SRIOV-GA.md

```diff
@@ -0,0 +1,179 @@
+both CNI config and runtimeConfig. In case of runtimeConfig, multus reads
+address from network annotation in pod and multus supply it as runtimeConfig.
+However, current Multus only supports passing IP configuration via CNI config.
+This proposal is to add runtimeConfig support in Multus CNI.
+
+##### Default Route Overwrite
+
+Every pod comes with a default route added via openshift-sdn. It is not
+possible to change this default route to secondary-onward interfaces.
+This proposal is to come up with a solution for Multus to handle default
+route overwriting.
+
+##### DHCP Configuration Guide
+
+DHCP IPAM plugin configuration is complicated due to its architecture,
+for example, DHCP IPAM requires 'DHCP CNI daemon' and 'DHCP server'.
+Currently we only provide simple document and there were several inquiries
+from user to enhance this document. This proposal is to provide DHCP
+configuration guide to help user understand DHCP CNI configuration and
+troubleshooting.
+
+##### Mac Address Pool Management
+
+Fixed MAC address can be specified via pod annotation for SR-IOV devices.
+However, user has to manually add MAC address for each SR-IOV interface
+in pod annotation which is error prone and inefficient. This proposal is
+to improve user experience by providing a MAC pool management tool which
+automatically assigns a MAC address in pod annotation.
+
+#### SR-IOV Operator
+
+Currently, the Custom Resources of SR-IOV Network Operator can only be
+validated by the declarative OpenAPI v3 schema validation rules defined
+in CRD and setting default value for CR is not supported. 
+
+To improve the user experience, an admission controller webhook for the
+CRD of SR-IOV network Operator shall be implemented. The admission controller
+webhook is expected to validate and set default value for the Custom Resource.
+
+The admission controller webhook and its RBAC shall be provisioned to
+openshift master nodes as a deployment by the SRIOV network operator.
+
+The admission controller webhook is expected to:
+- Set default value for fields of SriovNetwork CR, if not specified by user
+- Set default value for fields of SriovNetworkNodePolicy CR, if not specified
+  by user
+- Validate during SriovNetwork CR creation and updating request. The webhook
+  shall be able to validate following errors:
+  - `networkNamespace` refer to non-existing namespace
+- Validate during SriovNetworkNodePolicy CR creation and updating request
+  The webhook shall be able to validate following errors:
+  - The `numVFs` is larger than the maximum value that the NIC can support
+  - The `nicSelector` selects PFs which has already been covered by existing
+    Policy CRs
+  - The `resourceName` contains invalid characters
+  - `pfName` or `rootDevices` refers to non-existing device.
+
+
+#### SR-IOV QoS Flags
+
+Enable VF flags for security and QoS features.
+
+- Spoof Check
+- Trusted VF
+- Max rate limit
+- Min rate limit
+
+### Risks and Mitigations
+
+#### NUMA
```

Comment:
This should be part of GA now that we have 1.16 in 4.3.

Reply:
fixed